src/main.rs

```diff
@@ -0,0 +1,169 @@
+    println!("Answe for Q2:  {}\n",answer.into_i64());
+    Ok(())
+}
+async fn query3(connection: Connection,q: String)->std::io::Result<()>{    
+    let databases = DatabaseManager::new(connection.clone());
+    let session = Session::new(databases.get(TEST_DATABASE).await.unwrap(), Data).await.unwrap();
+    let transaction = session.transaction(Read).await.unwrap();
+    println!("::Q3::");
+    let mut answer_stream = transaction.query().match_(q.as_str()).unwrap();
+    while let Some(result) = answer_stream.next().await{
+        match result {
+            Ok(concept_map) => {
+                for (_, concept) in concept_map {
+                    if let Concept::Attribute(Attribute { value: Value::String(value), .. }) = concept {
+                        println!("{}",value);
+                    }
+                }
+            }
+            Err(err) => {
+                panic!("An error occurred fetching answers of a Match query: {err}")
+            }
+        }
+    }
+    Ok(())
+}
+
+async fn runIO(connection: Connection){
+    println!("\nQ1->What places could buy raw_food=$x ?\nQ2->Get count of non-vegetarian outlets with vegetarian specialities.\nQ3->Get count raw items sold at places with avg_rating more tha $r and has a dish using it as raw_ingredient with price greater than $p units.\nWhat query would you like to make? Enter 1,2 or 3.\n");
+    let qtype: i32=read!();
+    if qtype==1{
+        println!("Enter the name of raw_food x : ");
+        let inp: String=read!();
+        let x=inp.as_str();
+        let p0="match $rf isa raw_food, has name \"";
+        let p1="\";
+        $is_i (raw_food: $rf,dish: $d) isa is_ingredient;
+        $sl (seller: $m,product: $d) isa sells;
+        $m has name $n,has call_number $c;
+        get $n;";
+        let mut f="".to_owned();
+        f=f+p0+x+p1;
+        query1(connection,f).await.unwrap();
+    }else if qtype==2{
+        let p0="match 
+        $m2 isa menu, has is_vegetarian false,has name $n2;
+        $d2 isa dish, has is_vegetarian false;
+        $sp2 (restaurant: $m2,$d2) isa speciality;
+        get $m2;count;";
+        let mut f="".to_owned();
+        f=f+p0;
+        query2(connection,f).await.unwrap();
+    }else if qtype==3{
+        println!("Enter the avg_rating of Restraunt r : ");
+        let inp0: String=read!();
+        let r=inp0.as_str();
+        println!("Enter the threshold price for raw_ingredient p : ");
+        let inp1: String=read!();
+        let p=inp1.as_str();
+        let p0="match
+        $m3 isa menu, has avg_rating>";
+        let p1=",has name $mn3;
+        $d3 isa dish, has name $dn3;
+        $sl3 (seller: $m3,product: $d3) isa sells, has price $p3;
+        $p3>";
+        let p2=";
+        $rf3 isa raw_food,has name $rfn3;
+        $isn3 (raw_food: $rf3,$d3) isa is_ingredient;
+        get $rfn3;";
+        let mut f="".to_owned();
+        f=f+p0+r+p1+p+p2;
```

Comment:
You should look into use of the `format!` macro. This will help when concatenating strings. For example, we can construct owned `String`s out of `str`s like:
```rs
let f = format!("{p0}{r}{p1}");

Reply:
Yes, I looked into ```format!``` macro. It looks pretty neat now . Done